app/src/main/java/com/sdp13epfl2021/projmag/activities/SectionSelectionActivity.kt

```diff
@@ -0,0 +1,85 @@
+package com.sdp13epfl2021.projmag.activities
+
+import android.content.Intent
+import android.graphics.Color
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import androidx.recyclerview.widget.RecyclerView
+import com.sdp13epfl2021.projmag.R
+import com.sdp13epfl2021.projmag.activities.listenerClass.MakeRecyclerListItemsClickListenable
+import com.sdp13epfl2021.projmag.adapter.SectionAdapter
+import com.sdp13epfl2021.projmag.model.SectionBaseManager
+import java.io.Serializable
+
+class SectionSelectionActivity : AppCompatActivity() {
+    private val  chosenSections : MutableSet<String> = mutableSetOf()
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_section_selection)
+
+
+
+        val manager = SectionBaseManager()
+        val sectionDataset = manager.sectionList()
+        val sectionRecyclerView = findViewById<RecyclerView>(R.id.recycler_section_view)
+        val saveButton = findViewById<Button>(R.id.doneSectionButton)
+        sectionRecyclerView.adapter = SectionAdapter(this, sectionDataset)
+        sectionRecyclerView.setHasFixedSize(true)
+
+        handleListeningOnElementsOfSectionRecyclerView(sectionRecyclerView, sectionDataset)
+        saveButtonHandling(saveButton)
+
+
+
+
+    }
+    /**
+     * Function that han dles the click and long click of elements of the tag recycler view
+     *
+     * @param tagRecyclerView : the tag recycler view
+     * @param tagsDataset : the dataset of tags
+     */
+    private fun handleListeningOnElementsOfSectionRecyclerView(sectionRecyclerView: RecyclerView, sectionDataset:List<String>){
+        sectionRecyclerView.addOnItemTouchListener(
+            MakeRecyclerListItemsClickListenable(
+                this,
+                sectionRecyclerView,
+                object : MakeRecyclerListItemsClickListenable.OnItemClickListener {
+
+                    override fun onItemClick(view: View, position: Int) {
+                        val holder = sectionRecyclerView.findViewHolderForLayoutPosition(position) as SectionAdapter.SectionViewHolder
```

Comment:
It would be easier (and less lines of code) to just add a click listener in the SectionAdpater (same as ProjectAdapter),
and pass the already choosen section lists to the adapter constructor.